sources/packages/load-test/src/utils/auth/token-helpers.ts

```diff
@@ -0,0 +1,101 @@
+import http from "k6/http";
+import { AuthorizedParties } from "../sims-api/sims-api";
+import { KEY_CLOAK_TOKEN_URL } from "../../../config.env";
+import { decodeJWT } from "./jwt-helpers";
+import { TokenCacheResponse, UserPasswordCredential } from "./models";
+
+/**
+ * Default amount of time to a token be expired and be
+ * considered candidate to be renewed.
+ */
+const TOKEN_RENEWAL_SECONDS = 30;
+
+/**
+ * Caches the token already acquired.
+ */
+const tokenCache: Record<string, TokenCacheResponse> = {};
+
+/**
+ * Define if a token need to be renewed based in the jwt exp property.
+ * @param jwtExp exp token property that defines the expiration.
+ * @param maxSecondsToExpired Amount of seconds before the token expires that should be considered to renew it.
+ * For instance, if a token has 10min expiration and the parameter is defined as 30, the token will
+ * be renewed the result will be true if the token was acquired 9min30s ago or if it is already expired.
+ * @returns true if the token is expired or about to expire based on the property maxSecondsToExpired.
+ */
+function needRenewJwtToken(jwtExp: number, maxSecondsToExpired = 0): boolean {
```

Comment:
FMU, is `maxSecondsToExpired` the total amount of time since login?

Reply:
Just checking, does the comment need some improvement? Asking because if it is not we need to adapt it.
"Amount of seconds before the token expires that should be considered to renew it.
For instance, if a token has 10min expiration and the parameter is defined as 30, the token will
be renewed the result will be true if the token was acquired 9min30s ago or if it is already expired."